tests/utilities/test_cli.py

```diff
@@ -364,15 +369,17 @@ def test_lightning_cli_save_config_cases(tmpdir):
 
 
 def test_lightning_cli_config_and_subclass_mode(tmpdir):
```

Comment:
@mauvilsa

I checked out https://github.com/omni-us/jsonargparse/pull/78 locally to test it and everything seems to be working but this test.

See this repro:

```python
import os
from unittest import mock

from pytorch_lightning.utilities.cli import LightningCLI
from tests.helpers import BoringModel, BoringDataModule

def test_1():
    tmpdir = os.getcwd()
    input_config = {
        "subcommand": "fit",
        "fit": {
            "model": {"class_path": "tests.helpers.BoringModel"},
            "data": {
                "class_path": "tests.helpers.BoringDataModule",
                "init_args": {"data_dir": str(tmpdir)},
            },
            "trainer": {
                "default_root_dir": str(tmpdir),
                "max_epochs": 1,
                "weights_summary": None,
            },
        },
    }

    with mock.patch("sys.argv", ["any.py", "--config", str(input_config)]):
         LightningCLI(
            BoringModel, BoringDataModule, subclass_mode_model=True, subclass_mode_data=True
        )



def test_2():
    tmpdir = os.getcwd()
    input_config = {
        "model": {"class_path": "tests.helpers.BoringModel"},
        "data": {
            "class_path": "tests.helpers.BoringDataModule",
            "init_args": {"data_dir": str(tmpdir)},
        },
        "trainer": {
            "default_root_dir": str(tmpdir),
            "max_epochs": 1,
            "weights_summary": None,
        },
    }

    with mock.patch("sys.argv", ["any.py", "fit", "--config", str(input_config)]):
         LightningCLI(
            BoringModel, BoringDataModule, subclass_mode_model=True, subclass_mode_data=True
        )
````

Both functions fail with the message:

`jsonargparse.util.ParserError: Conflicting namespace base: fit.model`

Any ideas about how to fix it?

Reply:
I fixed it locally but getting the error below. Do you know what the problem could be?

```
Traceback (most recent call last):
  File "./cli_7508_conflicting_namespace_base.py", line 66, in <module>
    test_2()
  File "./cli_7508_conflicting_namespace_base.py", line 62, in test_2
    parser_kwargs={'error_handler': None}
  File "/home/mvillegas/repos/thirdparty/pytorch-lightning/pytorch_lightning/utilities/cli.py", line 297, in __init__
    self._run_subcommand(subcommand)
  File "/home/mvillegas/repos/thirdparty/pytorch-lightning/pytorch_lightning/utilities/cli.py", line 500, in _run_subcommand
    fn(**fn_kwargs)
  File "/home/mvillegas/repos/thirdparty/pytorch-lightning/pytorch_lightning/trainer/trainer.py", line 539, in fit
    model, train_dataloaders=train_dataloaders, val_dataloaders=val_dataloaders, datamodule=datamodule
  File "/home/mvillegas/repos/thirdparty/pytorch-lightning/pytorch_lightning/trainer/connectors/data_connector.py", line 104, in attach_data
    self.attach_datamodule(model, datamodule=datamodule)
  File "/home/mvillegas/repos/thirdparty/pytorch-lightning/pytorch_lightning/trainer/connectors/data_connector.py", line 144, in attach_datamodule
    if is_overridden(method, datamodule):
  File "/home/mvillegas/repos/thirdparty/pytorch-lightning/pytorch_lightning/utilities/model_helpers.py", line 45, in is_overridden
    raise ValueError("Expected a parent")
ValueError: Expected a parent
```

Reply:
It seems to be working for me, both the test and the repro snippet.

Thank you for looking into it!